src/test/java/org/opensearch/knn/index/RestoreSnapshotIT.java

```diff
@@ -0,0 +1,122 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index;
+
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.client.ResponseException;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.common.xcontent.support.XContentMapValues;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.test.rest.OpenSearchRestTestCase;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.*;
+import static org.opensearch.common.xcontent.XContentFactory.jsonBuilder;
+
+public class RestoreSnapshotIT extends OpenSearchRestTestCase {
+
+    private String index;
+    private String snapshot;
+
+    public void setupSnapshotRestore() throws Exception {
+        Request clusterSettingsRequest = new Request("GET", "/_cluster/settings");
+        clusterSettingsRequest.addParameter("include_defaults", "true");
+        Response clusterSettingsResponse = client().performRequest(clusterSettingsRequest);
+        Map<String, Object> clusterSettings = entityAsMap(clusterSettingsResponse);
+
+        @SuppressWarnings("unchecked")
+        List<String> pathRepos = (List<String>) XContentMapValues.extractValue("defaults.path.repo", clusterSettings);
+        assertThat(pathRepos, notNullValue());
+        assertThat(pathRepos, hasSize(1));
+
+        final String pathRepo = pathRepos.get(0);
+
+        index = "test-index";
+        snapshot = "snapshot-" + index;
+
+        // create index
+        XContentBuilder settings = jsonBuilder();
+        settings.startObject();
+        {
+            settings.startObject("settings");
+            settings.field(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1);
+            settings.field(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1);
+            settings.field(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), true);
+            settings.endObject();
+        }
+        settings.endObject();
+
+        Request createIndex = new Request("PUT", "/" + index);
+        createIndex.setJsonEntity(settings.toString());
+        createIndex.setOptions(allowTypesRemovalWarnings());
+        client().performRequest(createIndex);
+
+        // create repo
+        XContentBuilder repoConfig = JsonXContent.contentBuilder().startObject();
+        {
+            repoConfig.field("type", "fs");
+            repoConfig.startObject("settings");
+            {
+                repoConfig.field("compress", randomBoolean());
+                repoConfig.field("location", pathRepo);
+            }
+            repoConfig.endObject();
+        }
+        repoConfig.endObject();
+        Request createRepoRequest = new Request("PUT", "/_snapshot/repo");
+        createRepoRequest.setJsonEntity(repoConfig.toString());
+        client().performRequest(createRepoRequest);
```

Comment:
Same for this, refer [here](https://github.com/opensearch-project/OpenSearch/blob/2.x/test/framework/src/main/java/org/opensearch/test/rest/OpenSearchRestTestCase.java#L1108). Lets check other API calls as well and see if we can instead use the helper methods for those too (not commenting for each separately)